index.d.ts

```diff
@@ -0,0 +1,28 @@
+import { Linter, Rule } from 'eslint';
+
+declare const plugin: {
+  meta: {
+    name: string;
+    version: string;
+  };
+  environments: {
+    globals: {
+      globals: {
+        [key: string]: boolean;
+      };
+    };
+  };
+  configs: {
+    all: Linter.Config;
+    recommended: Linter.Config;
+    style: Linter.Config;
```

Comment:
issue: we should use this type to be compatible with `@types/eslint@9`

```suggestion
    all: Linter.LegacyConfig;
    recommended: Linter.LegacyConfig;
    style: Linter.LegacyConfig;
```